intro-js.md

```diff
@@ -0,0 +1,149 @@
+# Introduction
+Welcome to the weird and wonderful world of JavaScript development. 
+JavaScript, as a language, is used primarily in the development of web applications and websites. 
+
+## Why is JavaScript weird? 
+
+1. Lack of types
+
+JavaScript is a *loosely typed* or *dynamically typed* language
+
+```javascript
+var a = 42; //a is a number here
+var a = 'random' //a is a string here
+var a = true //a is a boolean here
+```
+
+This makes JavaScript development cumbersome and bug ridden, but there are improvements being made. 
+
+A solution to this has been provided through [TypeScript](https://www.typescriptlang.org/docs/home.html)
+
+Angular 2 utilizes TypeScript extensively.
+
+2. Lack of classes
+JavaScript did not have the concept of classes when it was first written.
+
+Classes have now been provided as part of [ECMAScript 2015](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes), but this is just syntactic sugar. 
+
+Example class
+```javascript 
+class Shape {
+    constructor(param1, param2) {
+        
+    }
+}
+```
+
+3. Functions are first class objects 
+
+This one is a little hard to wrap your heads around so we won't go into too much detail, but it is one of the most fundamental aspects of JavaScript. It is what makes JavaScript such a powerful language. 
+
+Basically, what this means is that functions can be constructed during execution, they can be passed around as variables or parameters to other functions and you can also return a function as the value of another function (what is referred to as a currying). 
```

Comment:
Meat is here, and since this is a very difficult topic to cover, I'm not sure if it's 100% necessary